project.py

```diff
@@ -193,6 +299,12 @@ def video_search(self, video):
 
 def main(inputFile):
     board = BoardSearcher()
+    board.numberOfSlides = 0
```

Comment:
@Foxiy This might work in Java, but even there it is considered to be a bad practice.

What I suggest is the use of keyword arguments in class `BoardSearcher` with sensible defaults (like for instance the ones you specified below).